core/trino-main/src/test/java/io/trino/block/BlockAssertions.java

```diff
@@ -596,4 +626,237 @@ public static RunLengthEncodedBlock createRLEBlock(long value, int positionCount
+    {
+        checkArgument(dictionary.getPositionCount() > 0, format("dictionary's positionCount %d is less than or equal to 0", dictionary.getPositionCount()));
+
+        int idsOffset = 0;
+        if (isView) {
+            idsOffset = min(RANDOM.nextInt(dictionary.getPositionCount()), 1);
+        }
+
+        int[] ids = IntStream.range(0, positionCount + idsOffset).map(i -> RANDOM.nextInt(max(dictionary.getPositionCount() / 10, 1))).toArray();
+        return new DictionaryBlock(idsOffset, positionCount, dictionary, ids, false, randomDictionaryId());
+    }
+
+    public static Block createRandomBlockForType(
+            Type type,
+            int positionCount,
+            float primitiveNullRate,
+            float nestedNullRate,
+            boolean createView,
+            List<Encoding> wrappings)
+    {
+        verifyNullRate(primitiveNullRate);
+        verifyNullRate(nestedNullRate);
+
+        Block block = null;
+
+        if (createView) {
+            positionCount *= 2;
+        }
+
+        if (type == BOOLEAN) {
+            block = createRandomBooleansBlock(positionCount, primitiveNullRate);
+        }
+        else if (type == BIGINT) {
+            block = createRandomLongsBlock(positionCount, primitiveNullRate);
+        }
+        else if (type == INTEGER || type == REAL) {
+            block = createRandomIntsBlock(positionCount, primitiveNullRate);
+        }
+        else if (type == SMALLINT) {
+            block = createRandomSmallintsBlock(positionCount, primitiveNullRate);
+        }
+        else if (type instanceof DecimalType) {
+            DecimalType decimalType = (DecimalType) type;
+            if (decimalType.isShort()) {
+                block = createRandomLongsBlock(positionCount, primitiveNullRate);
+            }
+            else {
+                block = createRandomLongDecimalsBlock(positionCount, primitiveNullRate);
+            }
+        }
+        else if (type == VARCHAR) {
+            block = createRandomStringBlock(positionCount, primitiveNullRate, MAX_STRING_SIZE);
+        }
+        else {
+            block = createRandomBlockForNestedType(type, positionCount, primitiveNullRate, nestedNullRate, createView, wrappings);
+        }
+
+        if (createView) {
+            positionCount /= 2;
+            int offset = positionCount / 2;
+            block = block.getRegion(offset, positionCount);
+        }
+
+        if (!wrappings.isEmpty()) {
+            block = wrapBlock(block, positionCount, wrappings);
+        }
+
+        return block;
+    }
+
+    private static Block createRandomBlockForNestedType(Type type, int positionCount, float primitiveNullRate, float nestedNullRate, boolean createView, List<Encoding> wrappings)
+    {
+        Block block;
+        // Build isNull and offsets of size positionCount
+        boolean[] isNull = null;
+        Set<Integer> nullPositions = null;
+        if (nestedNullRate > 0) {
```

Comment:
`chooseNullPositions should work with `nestedNullRate` so `if` here is not needed